### PR TITLE
feat: support pre-extracted frame sequence of video sample

### LINKED ISSUE #3215: VLM validation drops the batch dimension for tensor-valued hidden_states
## Description

`FinetuneRecipeForVLM._run_validation_epoch()` assumes
`out.hidden_states` is a tuple/list and selects `out.hidden_states[-1]`.
Some VLMs, including Qwen3.6, return the final hidden state directly as a
tensor with shape `[B, T, H]`. Indexing `[-1]` then removes the batch
dimension and passes `[T, H]` to `FusedLinearCrossEntropy` while labels remain
`[B, T]`.

The training path already handles both output layouts correctly with
`get_final_hidden_states(out)`, but the validation path does not.

Reproduced against current `main` commit
`622cc24ba9206eee9cde6067ca0a50d95dcc8b25`.

## Minimal reproduction

This runs on CPU and does not require model weights:

```python
from types import SimpleNamespace

import torch

from nemo_automodel.components.training.model_output_utils import get_final_hidden_states

out = SimpleNamespace(hidden_states=torch.zeros(1, 4, 8))
labels = torch.zeros(1, 4, dtype=torch.long)

current_validation = out.hidden_states[-1]
correct_extraction = get_final_hidden_states(out)

print("current validation:", current_validation.shape)
print("correct extraction:", correct_extraction.shape)
print("labels:", labels.shape)

assert current_validation.shape[:-1] == labels.shape
```

Run with:

```bash
python repro.py
```

Output:

```text
current validation: torch.Size([4, 8])
correct extraction: torch.Size([1, 4, 8])
labels: torch.Size([1, 4])
AssertionError
```

The same mismatch occurs in the real VLM validation path when
`FusedLinearCrossEntropy` receives tensor-valued Qwen3.6 hidden states.

## Expected behavior

Validation should preserve `[B, T, H]` for direct tensor outputs and should
still select the final element when `hidden_states` is a tuple/list.

## Proposed fix

Replace the validation-specific indexing with the helper already imported and
used by the training path:

```python
hidden_states=get_final_hidden_states(out)
```

I am opening a companion PR with this one-line fix and a CPU-only regression
test that drives `_run_validation_epoch()` with tensor-valued hidden states.

## Environment where originally observed

- DGX Spark, single GB10 GPU
- NeMo AutoModel `0.5.0+d02f49cb`
- NVIDIA NeMo AutoModel 26.06 container
- PyTorch `2.12.0a0`
- Transformers `5.8.1`
- Qwen3.6-35B-A3B VLM
- `FusedLinearCrossEntropy` validation path

### PR DESCRIPTION
**Pull request by GITsologun**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog

- Add specific line by line info of high level changes in this PR.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by GITsologun** [comment]
{
  "messages": [
    {
      "content": "<video>生成监控视频的描述信息",
      "role": "user"
    },
    {
      "content": "监xxx箱前……",
      "role": "assistant"
    }
  ],
  "videos": [
    [
      "/nvmxxx17/frame_0000.png",
      "/nvmxxx17/frame_0013.png",
      "…共16帧,间隔约12.8帧…",
      "/nvxxx17/frame_0192.png"
    ]
  ],
  "images": []
}

**Comment by GITsologun** [comment]
image

**Comment by GITsologun** [comment]
The codecov/patch failure looks like an artifact of incomplete coverage uploads: the head report only has 2 sessions vs 4 on base, and project coverage dropped ~16pp which a ~130-line patch cannot cause. The new branch in utils.py is fully covered by the 7 unit tests added in 54ce0180 (TestPreloadMedia frame-sequence cases). Could you re-trigger CI (/ok to test 54ce0180) so coverage re-uploads?
